frontend-react/src/hooks/UseSenderResource.ts

```diff
@@ -0,0 +1,98 @@
+import { useMemo } from "react";
+
+import { useSessionContext } from "../contexts/SessionContext";
+import { createRequestConfig } from "../network/api/NewApi";
+import OrganizationsAPI from "../network/api/OrganizationsApi";
+
+import useRequestConfig from "./network/UseRequestConfig";
+
+type SenderKeys = {
+    scope: string;
+    keys: {}[];
+};
+
+type Sender = {
+    allowDuplicates: boolean;
+    createdAt?: string;
+    createdBy?: string;
+    customerStatus: string;
+    format: string;
+    keys?: SenderKeys;
+    name: string;
+    organizationName: string;
+    primarySubmissionMethod?: string;
+    processingType: string;
+    schemaName: string;
+    senderType?: string;
+    topic: string;
+    version?: number;
+};
+
+export const useSenderResource = () => {
+    /* Access the session. */
+    const { memberships, oktaToken } = useSessionContext();
+    /* Create a stable config reference with useMemo(). */
+    const config = useMemo(
+        () =>
+            createRequestConfig<{ org: string }>(
+                OrganizationsAPI,
+                "senders",
```

Comment:
i am wondering if it would be easier to use the `api/organizations/<org>/senders/<sender name>` endpoint instead of the full list to alleviate some of the error messaging and checking of senders vs. memberships below... :thinking: 
unless, we show the user what `sender`s they have in our system...
just a thought :smile: 
this works well as is!

Reply:
ah, didn't know that was an option. I think that probably would simplify some things. I will look into it. What is the best way to see a list of available endpoints?

Reply:
there is no real "good" way to look at the list, unfortunately. 
there is the openapi.yml file, but i think that is out of date.
and i think the Azure functions blurts out all of the endpoints when it starts up